river/cluster/clustream.py

```diff
@@ -0,0 +1,363 @@
+import math
+import sys
+
+from river import base, cluster
+from river.cluster.clustream_kernel import ClustreamKernel
+
+
+class Clustream(base.Clusterer):
+    """Clustream
+
+    It maintains statistical information about the data using micro-clusters.
+    These micro-clusters are temporal extensions of cluster feature vectors.
+    The micro-clusters are stored at snapshots in time following a pyramidal
+    pattern. This pattern allows to recall summary statistics from different
+    time horizons in [1]_.
+
+    Parameters
+    ----------
+    seed
+       If int, seed is the seed used by the random number generator;
+       If RandomState instance, seed is the random number generator;
+       If None, the random number generator is the RandomState instance used
+       by `np.random`.
+       It is used in the `KMeans` algorithm for the offline clustering, which is the native `KMneans` implemented in `River`.
+
+    time_window
+      The rang of the window
+      if the current time is T and the time window is h, we should only consider
+      about the data that arrived within the period (T-h,T)
+
+    max_kernels
+      The Maximum number of micro kernels to use
+
+    kernel_radius_factor
+      Multiplier for the kernel radius
+      When deciding to add a new data point to a micro-cluster, the maximum boundary
+      is defined as a factor of the kernel_radius_factor of the RMS deviation of the
+      data points in the micro-cluster from the centroid
+
+    number_of_clusters
+        the clusters returned by the Kmeans algorithm using the summaries statistics
+
+    kwargs
+        Other parameters passed to the incremental kmeans at `river.cluster.KMeans`
+
+    Attributes
+    ----------
+    centers : dict
+        Central positions of each cluster.
+
+    Examples
+    --------
+
+    In the following implementation, due to the limited number of points passed through the model,
+    max_kernels and time_window are set relatively low. Moreover, all points are learnt before any predictions are made.
+    The halflife are set at 0.4, passing from cluster.KMeans, to test the functionality of keyword arguments.
+
+    >>> from river import cluster
+    >>> from river import stream
+
+    >>> X = [
+    ...     [1, 2],
+    ...     [1, 4],
+    ...     [1, 0],
+    ...     [4, 2],
+    ...     [4, 4],
+    ...     [4, 0]
+    ... ]
+
+    >>> clustream_test = cluster.Clustream(time_window = 1, max_kernels = 3, number_of_clusters = 2, seed = 0, halflife = 0.4)
```

Comment:
Don't use spaces around `=`. You can put the declaration on multiple lines to improve readability.

Reply:
Resolved.